consensus/istanbul/backend/handler.go

```diff
@@ -117,6 +118,17 @@ func (sb *Backend) NewChainHead() error {
 		} else {
 			sb.logger.Info("Validators Election Results: Node IN ValidatorSet")
 		}
+
+		// Set the randomness for proposer selection ordering from the block.
```

Comment:
I put it here rather than where we create/fetch the snapshot because at that location we are not guaranteed to have the EVM state. But it's likely I don't understand something here fully.

Reply:
It's also quite possible that I don't understand.

Would changing the `valSet` here actually persist any changes though? It doesn't seem like it would update `sb.core.valSet`, which is what I would have thought we would need to change

Reply:
You can see in `core/core.go` in `startNewRound` that the `valSet` that we use to calculate the proposer is pulled from the backend snapshots, so if we don't write the randomness to the snapshot I wouldn't expect us to actually calculate the proposer with that randomness.

It would be good to confirm that this PR is working in the governance end-to-end tests by checking the order of proposers